src/test/java/cz/upce/fei/inptp/zz/entity/TimeSlotTest.java

```diff
@@ -20,39 +20,93 @@ public void testSameSlotsShouldBeEqual() {
-        int hour_a = 8;
-        int hour_b = 12;
-        int duration = 7;
+        int startHour_a = 8;
+        int startHour_b = 12;
+        int endHour = 13;
         
-        TimeSlot ta = new TimeSlot(day, hour_a, duration);
-        TimeSlot tb = new TimeSlot(day, hour_b, duration);
+        TimeSlot ta = new TimeSlot(day, startHour_a, endHour);
+        TimeSlot tb = new TimeSlot(day, startHour_b, endHour);
         
         assertNotEquals(ta, tb);
     }
     
     @Test
-    public void testDifferentDurationsShouldNotBeEqual() {
+    public void testDifferentEndHoursShouldNotBeEqual() {
         Day day = Day.Friday;
-        int hour = 8;
-        int duration_a = 7;
-        int duration_b = 12;
+        int startHour = 8;
+        int endHour_a = 9;
+        int endHour_b = 12;
         
-        TimeSlot ta = new TimeSlot(day, hour, duration_a);
-        TimeSlot tb = new TimeSlot(day, hour, duration_b);
+        TimeSlot ta = new TimeSlot(day, startHour, endHour_a);
+        TimeSlot tb = new TimeSlot(day, startHour, endHour_b);
         
         assertNotEquals(ta, tb);
     }
     
+    @Test(expected = IllegalArgumentException.class)
+    public void testStartHourIsNotCorrect() {
+        TimeSlot t = new TimeSlot(Day.Monday, 0, 15);
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void testEndHourIsNotCorrect() {
+        TimeSlot t = new TimeSlot(Day.Monday, 8, 22);
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void testEndHourIsLessThanStartHourThrowsException() {
+        TimeSlot t = new TimeSlot(Day.Monday, 10, 9);
+    }
+    
+    @Test
+    public void testSameTimeSlotsAreOverlapping() {
+        TimeSlot t_a = new TimeSlot(Day.Monday, 9, 11);
+        TimeSlot t_b = new TimeSlot(Day.Monday, 9, 11);
+
+        assertTrue(t_a.isOverlappingWithOtherTimeslot(t_b));
+    }
+
+    @Test
+    public void testTimeSlotsWithSameDayAndStartHourAreOverlapping() {
+        TimeSlot t_a = new TimeSlot(Day.Monday, 9, 11);
+        TimeSlot t_b = new TimeSlot(Day.Monday, 9, 12);
+
+        assertTrue(t_a.isOverlappingWithOtherTimeslot(t_b));
+    }
+
+    @Test
+    public void testTimeSlotsWithSameDayAndEndHourAreOverlapping() {
+        TimeSlot t_a = new TimeSlot(Day.Monday, 10, 12);
+        TimeSlot t_b = new TimeSlot(Day.Monday, 9, 12);
+
+        assertTrue(t_a.isOverlappingWithOtherTimeslot(t_b));
+    }
+
+    @Test
+    public void testTimeSlotsWithDifferentStartAndEndHoursAreOverlapping() {
+        TimeSlot t_a = new TimeSlot(Day.Monday, 10, 13);
+        TimeSlot t_b = new TimeSlot(Day.Monday, 11, 12);
+
+        assertTrue(t_a.isOverlappingWithOtherTimeslot(t_b));
+    }
+
+    @Test
+    public void testTimeSlotsWithDifferentDayAreNotOverlapping() {
+        TimeSlot t_a = new TimeSlot(Day.Friday, 10, 13);
+        TimeSlot t_b = new TimeSlot(Day.Monday, 10, 13);
+
+        assertFalse(t_a.isOverlappingWithOtherTimeslot(t_b));
+    }
```

Comment:
Please add also checks for states:
`TA:---XXXX------------------`
`TB:--------------XXXXXX-----`
And also reversed TB-TA.